Build error when openPMD_USE_HDF5=OFF for the examples

**Describe the bug**
When I tried to disable support for HDF5, the build process still looks for HDF5 and tries to link to the library for the examples.

**To Reproduce**

```bash
# in directory $HOME/src/openPMD-api
mkdir build && cd build

CC=mpicc CXX=mpicxx cmake .. -DopenPMD_USE_MPI=ON -DopenPMD_USE_ADIOS2=ON -DopenPMD_USE_HDF5=OFF -DopenPMD_USE_ADIOS1=ON -DCMAKE_CXX_STANDARD=14
```
```
Consolidate compiler generated dependencies of target 8a_benchmark_write_parallel
[ 54%] Linking CXX executable bin/8a_benchmark_write_parallel
lib/libopenPMD.so: undefined reference to `H5Dset_extent'
lib/libopenPMD.so: undefined reference to `H5Pcreate'
lib/libopenPMD.so: undefined reference to `H5Aexists'
lib/libopenPMD.so: undefined reference to `H5T_NATIVE_INT8_g'
lib/libopenPMD.so: undefined reference to `H5Tis_variable_str'
lib/libopenPMD.so: undefined reference to `H5Oopen'
lib/libopenPMD.so: undefined reference to `H5Tget_size'
lib/libopenPMD.so: undefined reference to `H5Fopen'
lib/libopenPMD.so: undefined reference to `H5T_NATIVE_INT_g'
lib/libopenPMD.so: undefined reference to `H5check_version'
lib/libopenPMD.so: undefined reference to `H5Pset_coll_metadata_write'
lib/libopenPMD.so: undefined reference to `H5Gget_info'
lib/libopenPMD.so: undefined reference to `H5Awrite'
lib/libopenPMD.so: undefined reference to `H5Aclose'
lib/libopenPMD.so: undefined reference to `H5Dget_create_plist'
lib/libopenPMD.so: undefined reference to `H5Acreate2'
lib/libopenPMD.so: undefined reference to `H5Ldelete'
lib/libopenPMD.so: undefined reference to `H5Dget_space'
lib/libopenPMD.so: undefined reference to `H5Pset_all_coll_metadata_ops'
lib/libopenPMD.so: undefined reference to `H5Pget_chunk'
lib/libopenPMD.so: undefined reference to `H5Gcreate2'
lib/libopenPMD.so: undefined reference to `H5Tget_member_name'
lib/libopenPMD.so: undefined reference to `H5Aopen'
lib/libopenPMD.so: undefined reference to `H5T_NATIVE_LLONG_g'
lib/libopenPMD.so: undefined reference to `H5Tenum_insert'
```

**Expected behavior**
If I disabled the HDF5, I would expect it not to look for the library files and build only the ADIOS support.

**Software Environment**
 - version of openPMD-api: `9170612465e2eac3ad5258e6c746b59dda64a369`
 - installed openPMD-api via: source
 - operating system: 
 - machine: Summit
 - version of HDF5: hdf5/1.10.4
 - version of ADIOS1: adios/1.13.1-py2
 - version of ADIOS2: adios2/2.7.0
 - name and version of MPI: spectrum-mpi/10.3.1.2-20200121

--- a/.gitignore
+++ b/.gitignore
@@ -88,17 +88,6 @@ Network Trash Folder
 Temporary Items
 .apdisk
 
-#########
-# CMake #
-#########
-CMakeCache.txt
-CMakeFiles
-CMakeScripts
-cmake_install.cmake
-install_manifest.txt
-compile_commands.json
-CTestTestfile.cmake
-
 ####################
 # Package Managers #
 ####################

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -19,6 +19,18 @@ if(POLICY CMP0074)
 endif()
 
 
+# No in-Source builds #########################################################
+#
+# In-source builds clutter up the source directory and lead to mistakes with
+# generated includes
+if(openPMD_SOURCE_DIR STREQUAL openPMD_BINARY_DIR)
+  message(FATAL_ERROR "In-source builds are not possible. "
+          "Please remove the CMakeFiles/ directory and CMakeCache.txt file. "
+          "Then run CMake in a temporary build directory. "
+          "Learn more: https://hsf-training.github.io/hsf-training-cmake-webpage/02-building/index.html")
+endif()
+
+
 # Project structure ###########################################################
 #
 # temporary build directories

